core/dbt/adapters/base/impl.py

```diff
@@ -1327,7 +1327,7 @@ def catch_as_completed(
         elif isinstance(exc, KeyboardInterrupt) or not isinstance(exc, Exception):
             raise exc
         else:
-            warn_or_error(f"Encountered an error while generating catalog: {str(exc)}")
+            warn_or_error(CatalogGenerationError(exc=str(exc)))
```

Comment:
Nit: this paradigm of stringifying arguments to event types now has the option of adding [this class](https://github.com/dbt-labs/dbt-core/blob/8145eed603266951ce35858f7eef3836012090bd/core/dbt/events/base_types.py#L112) as a subclass to the event class definition.

Reply:
That class is super cool.  However, it is causing some issues because protobuf requires us to define the type of each field being passed in in a very narrow scope (types cannot be classes ie Exception).  So `exc` comes in as an `Exception` and then does become a string in the message but that's not enough since proto expects the field itself to be a string already.